app/controllers/api/base_controller/renderer.rb

```diff
@@ -149,6 +153,185 @@ def filter_resource(target, type, klass)
+
+          @reconfig_values[:disk_add] = @req.options[:disk_add]
+          @reconfig_values[:disk_resize] = @req.options[:disk_resize]
+          @reconfig_values[:cdrom_connect] = @req.options[:cdrom_connect]
+          @reconfig_values[:cdrom_disconnect] = @req.options[:cdrom_disconnect]
+          vmdisks = []
+          vmcdroms = []
+          @req.options[:disk_add]&.each do |disk|
+            adsize, adunit = reconfigure_calculations(disk[:disk_size_in_mb])
+            vmdisks << {:hdFilename          => disk[:disk_name],
+                        :hdType              => disk[:thin_provisioned] ? 'thin' : 'thick',
+                        :hdMode              => disk[:persistent] ? 'persistent' : 'nonpersistent',
+                        :hdSize              => adsize.to_s,
+                        :hdUnit              => adunit,
+                        :new_controller_type => disk[:new_controller_type].to_s,
+                        :cb_dependent        => disk[:dependent],
+                        :cb_bootable         => disk[:bootable],
+                        :add_remove          => 'add'}
+          end
+          
+          reconfig_item = Vm.find(reconfigure_ids)
+          if reconfig_item
+            reconfig_item.first.hardware.disks.each do |disk|
+              next if disk.device_type != 'disk'
+
+              removing = ''
+              delbacking = false
+              if disk.filename && @req.options[:disk_remove]
+                @req.options[:disk_remove].each do |remdisk|
+                  if remdisk[:disk_name] == disk.filename
+                    removing = 'remove'
+                    delbacking = remdisk[:delete_backing]
+                  end
+                end
+              end
+              dsize, dunit = reconfigure_calculations(disk.size / (1024 * 1024))
+              vmdisk = {:hdFilename     => disk.filename,
+                        :hdType         => disk.disk_type.to_s,
+                        :hdMode         => disk.mode.to_s,
+                        :hdSize         => dsize.to_s,
+                        :hdUnit         => dunit.to_s,
+                        :delete_backing => delbacking,
+                        :cb_bootable    => disk.bootable,
+                        :add_remove     => removing}
+              vmdisks << vmdisk
+            end
+            cdroms = reconfig_item.first.hardware.cdroms
+            if cdroms.present?
+              vmcdroms = build_request_cdroms_list(cdroms)
+            end
+          end
+          @reconfig_values[:disks] = vmdisks
+          @reconfig_values[:cdroms] = vmcdroms
+        end
+    
+        @reconfig_values[:cb_memory] = false  #if @request_id != new we need to change false
+        @reconfig_values[:cb_cpu] = false #if @request_id != new we need to change false
+        @reconfig_values
+      end
+
+      def get_reconfig_info(reconfigure_ids)
+        @reconfigureitems = Vm.find(reconfigure_ids).sort_by(&:name)
```

Comment:
NOTE this is bypassing RBAC checks, don't do direct queries like this that is what resource_search is for.